libcxx/test/libcxx/transitive_includes/diff.py

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env python
+# ===----------------------------------------------------------------------===##
+#
+# Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# ===----------------------------------------------------------------------===##
+
+import argparse
+import os
+import sys
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="""Diff two files.""",
+    )
+    parser.add_argument("file1", default=None)
+    parser.add_argument("file2", default=None)
+    args = parser.parse_args()
+
+    def doread(f):
```

Comment:
This should produce much better output:

```
from pathlib import Path
import argparse, difflib, sys

def main():
  parser = argparse.ArgumentParser(description='Diff two files')
  parser.add_argument('a', help='First file to diff')
  parser.add_argument('b', help='Second file to diff')
  args = parser.parse_args()

  # open two files & spray the diff to stdout; return false iff a proble
  a = Path(args.a).read_text().splitlines(keepends=True)
  b = Path(args.b).read_text().splitlines(keepends=True)
  has_diff = False
  for line in difflib.context_diff(a, b, args.a, args.b):
    has_diff = True
    print(line, end='', file=sys.stderr)

  if has_diff:
    print(f'Files differ! [{args.a}, {args.b}]', file=sys.stderr)
    sys.exit(1)


main()

```

Reply:
Ack, I can change to that. TBH this whole patch is a bit throwaway, my motivation was to unblock your work on the docker-in-docker setup ASAP, and then investigate why Lit's builtin `diff` command didn't work properly.

Reply:
Oh, sorry actually I missed that you were using `difflib`. I was using `difflib` in an earlier version of this patch and it didn't work, so I dumbed it down to comparing the lines naively.

Reply:
I don't think difflib is at fault here, but I'm OK with this change then.